crates/exex-test-utils/src/lib.rs

```diff
@@ -0,0 +1,248 @@
+    },
+    BuilderContext, Node, NodeAdapter, RethFullAdapter,
+};
+use reth_node_core::node_config::NodeConfig;
+use reth_node_ethereum::{
+    node::{EthereumNetworkBuilder, EthereumPayloadBuilder},
+    EthEngineTypes, EthEvmConfig,
+};
+use reth_payload_builder::noop::NoopPayloadBuilderService;
+use reth_primitives::{ChainSpec, Head, SealedBlockWithSenders, MAINNET};
+use reth_provider::{
+    providers::BlockchainProvider, test_utils::create_test_provider_factory_with_chain_spec,
+    BlockReader, ProviderFactory,
+};
+use reth_tasks::TaskManager;
+use reth_transaction_pool::test_utils::{testing_pool, TestPool};
+use std::{
+    fmt::Debug,
+    future::{poll_fn, Future},
+    sync::Arc,
+    task::Poll,
+};
+use tokio::sync::mpsc::{Sender, UnboundedReceiver};
+
+/// A test [`PoolBuilder`] that builds a [`TestPool`].
+#[derive(Debug, Default, Clone, Copy)]
+#[non_exhaustive]
+pub struct TestPoolBuilder;
+
+impl<Node> PoolBuilder<Node> for TestPoolBuilder
+where
+    Node: FullNodeTypes,
+{
+    type Pool = TestPool;
+
+    async fn build_pool(self, _ctx: &BuilderContext<Node>) -> eyre::Result<Self::Pool> {
+        Ok(testing_pool())
+    }
+}
+
+/// A test [`ExecutorBuilder`] that builds a [`MockExecutorProvider`].
+#[derive(Debug, Default, Clone, Copy)]
+#[non_exhaustive]
+pub struct TestExecutorBuilder;
+
+impl<Node> ExecutorBuilder<Node> for TestExecutorBuilder
+where
+    Node: FullNodeTypes,
+{
+    type EVM = EthEvmConfig;
+    type Executor = MockExecutorProvider;
+
+    async fn build_evm(
+        self,
+        _ctx: &BuilderContext<Node>,
+    ) -> eyre::Result<(Self::EVM, Self::Executor)> {
+        let evm_config = EthEvmConfig::default();
+        let executor = MockExecutorProvider::default();
+
+        Ok((evm_config, executor))
+    }
+}
+
+/// A test [`Node`].
+#[derive(Debug, Default, Clone, Copy)]
+#[non_exhaustive]
+pub struct TestNode;
+
+impl NodeTypes for TestNode {
+    type Primitives = ();
+    type Engine = EthEngineTypes;
+}
+
+impl<N> Node<N> for TestNode
+where
+    N: FullNodeTypes<Engine = EthEngineTypes>,
+{
+    type ComponentsBuilder = ComponentsBuilder<
+        N,
+        TestPoolBuilder,
+        EthereumPayloadBuilder,
+        EthereumNetworkBuilder,
+        TestExecutorBuilder,
+    >;
+
+    fn components_builder(self) -> Self::ComponentsBuilder {
+        ComponentsBuilder::default()
+            .node_types::<N>()
+            .pool(TestPoolBuilder::default())
+            .payload(EthereumPayloadBuilder::default())
+            .network(EthereumNetworkBuilder::default())
+            .executor(TestExecutorBuilder::default())
+    }
+}
+
+type TmpDB = Arc<TempDatabase<DatabaseEnv>>;
+type Adapter = NodeAdapter<
+    RethFullAdapter<TmpDB, TestNode>,
+    <<TestNode as Node<FullNodeTypesAdapter<TestNode, TmpDB, BlockchainProvider<TmpDB>>>>::ComponentsBuilder as NodeComponentsBuilder<
+        RethFullAdapter<TmpDB, TestNode>,
+    >>::Components,
+>;
```

Comment:
we should do smth about this,
not loving how this looks -.-

Reply:
took me a while to get all the types right :D